driver/node.go

```diff
@@ -447,8 +472,101 @@ func (d *Driver) NodeExpandVolume(ctx context.Context, req *csi.NodeExpandVolume
 	return &csi.NodeExpandVolumeResponse{}, nil
 }
 
-// getDiskSource returns the absolute path of the attached volume for the given
+func (d *Driver) nodePublishVolumeForFileSystem(req *csi.NodePublishVolumeRequest, mountOptions []string, log *logrus.Entry) error {
+	source := req.StagingTargetPath
+	target := req.TargetPath
+
+	mnt := req.VolumeCapability.GetMount()
+	for _, flag := range mnt.MountFlags {
+		mountOptions = append(mountOptions, flag)
+	}
+
+	fsType := "ext4"
+	if mnt.FsType != "" {
+		fsType = mnt.FsType
+	}
+
+	mounted, err := d.mounter.IsMounted(target)
+	if err != nil {
+		return err
+	}
+
+	log.WithFields(logrus.Fields{
+		"source_path":   source,
+		"volume_mode":   volumeModeFilesystem,
+		"fs_type":       fsType,
+		"mount_options": mountOptions,
+	})
+
+	if !mounted {
+		log.Info("mounting the volume")
+		if err := d.mounter.Mount(source, target, fsType, mountOptions...); err != nil {
+			return status.Error(codes.Internal, err.Error())
+		}
+	} else {
+		log.Info("volume is already mounted")
+	}
+
+	return nil
+}
+
+func (d *Driver) nodePublishVolumeForBlock(req *csi.NodePublishVolumeRequest, mountOptions []string, log *logrus.Entry) error {
+	var ok bool
+	var volumeName string
+	if volumeName, ok = req.GetPublishContext()[d.publishInfoVolumeName]; !ok {
```

Comment:
Sorry I didn't see this on the first pass, but if we move the`if` to the next line like

```go
volumeName, ok = req.GetPublishContext()[d.publishInfoVolumeName]
if !ok {
  [...]
}
```

then we don't need the preceding variable declarations for `ok` and `volumeName`.

Reply:
:+1:, I think that is a better approach